### PR TITLE
Avoid unused member warning in ThreadPool

### PR DESCRIPTION
**Pull request by ChuanqiXu9**
## Why

Closing #109

### CODE REVIEW

File: async_simple/util/ThreadPool.h [hunk 1]
```diff
@@ -128,6 +128,9 @@ inline ThreadPool::ThreadPool(size_t threadNum, bool enableWorkSteal,
     std::vector<uint32_t> cpu_ids;
     if (_enableCoreBindings)
         getCurrentCpus(cpu_ids);
+#else
+    // Avoid unused member warning.
+    (void)_enableCoreBindings;
```
Comment (RainMark) [inline async_simple/util/ThreadPool.h:133]: can we use `[[maybe_unused]]` attribute for member field?

Comment (ChuanqiXu9) [reply async_simple/util/ThreadPool.h]: Good idea! Done.

Comment (ChuanqiXu9) [reply async_simple/util/ThreadPool.h:134]: It would be ignored by GCC and cause another error. I would use the original style.

Comment (RainMark) [reply async_simple/util/ThreadPool.h:134]: ok

**Review by RainMark — APPROVED** [review]

**Pull request merged**